CraftTweaker2-API/src/main/java/crafttweaker/api/recipes/IRecipeManager.java

```diff
@@ -175,6 +175,14 @@ public interface IRecipeManager {
      */
     @ZenMethod
     void removeShapeless(IIngredient output, @Optional IIngredient[] ingredients, @Optional boolean wildcard);
+
+    /**
+     * Remove all recipes that contain specified ingredient as input
+     *
+     * @param input The input ingredient
+     */
+    @ZenMethod
+    void removeByInput(IIngredient input);
```

Comment:
While I don't think anyone has implemented their own IRecipeManager, I still think it would be prudent to have a default implementation just incase